src/windows.rs

```diff
@@ -0,0 +1,306 @@
+            });
+            return;
+        }
+
+        let (watching_file,dir_target) = {
+            if path.is_dir() {
+                (false,path.clone())
+            } else {
+                // emulate file watching by watching the parent directory
+                (true,path.parent().unwrap().to_path_buf())
+            }
+        };
+
+        let encoded_path: Vec<u16> = dir_target.as_os_str().encode_wide().chain(Some(0)).collect();
+        let handle;
+        unsafe {
+            handle = kernel32::CreateFileW(
+                encoded_path.as_ptr(),
+                winnt::FILE_LIST_DIRECTORY,
+                winnt::FILE_SHARE_READ | winnt::FILE_SHARE_DELETE | winnt::FILE_SHARE_WRITE,
+                ptr::null_mut(),
+                fileapi::OPEN_EXISTING,
+                winbase::FILE_FLAG_BACKUP_SEMANTICS | winbase::FILE_FLAG_OVERLAPPED,
+                ptr::null_mut());
+
+            if handle == INVALID_HANDLE_VALUE {
+                let err = if watching_file {
+                    Err(Error::Generic("You attempted to watch a single file, but parent directory could not be opened.".to_owned()))
+                } else {
+                    // TODO: Call GetLastError for better error info?
+                    Err(Error::PathNotFound)
+                };
+                let _ = self.tx.send(Event {
+                    path: None,
+                    op: err
+                });
+                return;
+            }
+        }
+        let wf = if watching_file {
+            Some(path.clone())
+        } else {
+            None
+        };
+        let rd = ReadData {
+            dir: dir_target,
+            file: wf
+        };
+        self.watches.insert(path.clone(), handle);
+        start_read(&rd, &self.tx, handle);
+    }
+
+    fn remove_watch(&mut self, path: PathBuf) {
+        if let Some(handle) = self.watches.remove(&path) {
+            unsafe {
+                close_handle(handle);
+            }
+        }
+    }
+}
+
+unsafe fn close_handle(handle: HANDLE) {
+    // TODO: Handle errors
+    kernel32::CancelIo(handle);
+    kernel32::CloseHandle(handle);
+}
+
+fn start_read(rd: &ReadData, tx: &Sender<Event>, handle: HANDLE) {
+    let mut request = Box::new(ReadDirectoryRequest {
+        tx: tx.clone(),
+        handle: handle,
+        buffer: [0u8; BUF_SIZE as usize],
+        data: rd.clone()
+    });
+
+    let flags = winnt::FILE_NOTIFY_CHANGE_FILE_NAME
+              | winnt::FILE_NOTIFY_CHANGE_DIR_NAME
+              | winnt::FILE_NOTIFY_CHANGE_ATTRIBUTES
+              | winnt::FILE_NOTIFY_CHANGE_SIZE
+              | winnt::FILE_NOTIFY_CHANGE_LAST_WRITE
+              | winnt::FILE_NOTIFY_CHANGE_CREATION
+              | winnt::FILE_NOTIFY_CHANGE_SECURITY;
+
+    let monitor_subdir = if (&request.data.file).is_none() {
+      1
+    } else {
+      0
+    };
+
+    unsafe {
+        let mut overlapped: Box<OVERLAPPED> = Box::new(mem::zeroed());
```

Comment:
The only time these get deallocated is when handling an event. When you shut down the thread you don't wait for the final set of callbacks to occur so you can cleanup the stuff, thus leaking any of these that were still in progress.